### PR TITLE
✨ Cancel hook when a git message is already set

### PR DESCRIPTION
**Pull request by Jeremie-Chauvel**
## Description

Hi @carloscuesta ,

At first I just wanted to build end to end tests over @nathan818fr [pull request](https://github.com/carloscuesta/gitmoji-cli/pull/366). However there are numerous difficulties in doing a true E2E testing of a CLI app. The deal breaker for gitmoji being that inquirer doesn't handle piping to answer. I also tried to mock some libs like inquirer to avoid the prompt but still use a classic `git commit` flow to trigger the `prepare-commit-msg` hook without success.

My last idea was to use some remotely controllable terminal to deal with the interactivity of inquirer (similar in principle to cypress for web app), but except [kitty terminal](https://sw.kovidgoyal.net/kitty/#remote-control), I don't know another terminal that implement this feature and as kitty is still recent, I figured it wasn't a good fit since stability isn't reached yet and checked out this idea as well.

Since I couldn't figure out a way of doing such e2e tests in a reliable and cross platform way, I choose to go back to the source of the problem:
- The hook is always triggered when committing even on rebase, cherry-pick, amend, etc.
- Implementing the logic of not triggering the hook when the git message is already defined should be done in a plateform agnostic way, which as shown by https://github.com/carloscuesta/gitmoji-cli/issues/330 is no small task.

Finally, I tried a small proof of concept for a pure js implementation in gitmoji and figured out it was quite an easy and elegant solution with the added benefit of being easy to test.

## Issues

Issue: https://github.com/carloscuesta/gitmoji-cli/issues/330
Issue: https://github.com/carloscuesta/gitmoji-cli/issues/397
blocked pr: https://github.com/carloscuesta/gitmoji-cli/pull/366

## Tests

- [x] All tests passed.

### CODE REVIEW

**Comment by Jeremie-Chauvel** [comment]
Hi @carloscuesta,
do you think you could take a look ?

**Comment by carloscuesta** [comment]
Hey @Jeremie-Chauvel 

Sure I'm taking a look right now. Sorry for the delay I was on vacation

Thanks!

File: src/commands/commit/withHook/index.js [hunk 1]
```diff
@@ -26,4 +26,15 @@ export const registerHookInterruptionHandler = () => {
   })
 }
 
+export const cancelIfMessageIsAlreadySet = () => {
+  if (
+    fs.existsSync(process.argv[3]) &&
+    fs.lstatSync(process.argv[3]).isFile() &&
+    !!fs.readFileSync(process.argv[3], 'utf8')
```
Comment (carloscuesta) [inline src/commands/commit/withHook/index.js:33]: Do you think we can wrap the `process.argv[3]` on a variable and give it a explanation name ?

The name should relate why we use `3` from the `process.argv`

Comment (Jeremie-Chauvel) [reply src/commands/commit/withHook/index.js]: opted for a basic `commitMessageFilePath`

Comment (carloscuesta) [reply src/commands/commit/withHook/index.js]: That works perfect!

**Comment by carloscuesta** [comment]
Hey @Jeremie-Chauvel `3.2.8` which includes your change has been released and published to NPM!

**Pull request merged**